com.stencyl.Game.json

```diff
@@ -10,46 +10,71 @@
         "--share=network",
         "--device=all",
-        "--filesystem=host",
-        "--extra-data=Stencyl-64-full.tar.gz:c27d00dc83b5f443c35d785a0d22340f35c1cdb289c8cfe30417963475d4af29:207725021::http://mario.stencyl.net/public/Stencyl-64-full.tar.gz"
+        "--filesystem=host"
     ],
     "modules": [
         {
-            "name": "resources",
-            "no-autogen": true,
+            "name": "stencyl",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -D firefox /app/bin/firefox",
+                "install -D stencyl.sh /app/bin/stencyl",
+                "install -D apply_extra /app/bin/apply_extra",
+                "install -Dm644 StencylWorks.prefs /app/StencylPrefs/StencylWorks.prefs",
+                "install -Dm644 com.stencyl.Game.desktop /app/share/applications/com.stencyl.Game.desktop",
+                "install -Dm644 com.stencyl.Game.appdata.xml /app/share/appdata/com.stencyl.Game.appdata.xml",
+                "install -Dm644 com.stencyl.Game-64.png /app/share/icons/hicolor/64x64/apps/com.stencyl.Game.png",
+                "install -Dm644 com.stencyl.Game-128.png /app/share/icons/hicolor/128x128/apps/com.stencyl.Game.png",
+                "install -Dm644 com.stencyl.Game-512.png /app/share/icons/hicolor/512x512/apps/com.stencyl.Game.png"
+            ],
             "sources": [
+                {
+                    "type": "script",
+                    "dest-filename": "apply_extra",
+                    "commands": [
+                        "tar -xzvf Stencyl-64-full.tar.gz",
+                        "rm -rf Stencyl-64-full.tar.gz"
+                    ]
+                },
+                {
+                    "type": "extra-data",
+                    "filename": "Stencyl-64-full.tar.gz",
+                    "only-arches": ["x86_64"],
+                    "url": "http://mario.stencyl.net/public/Stencyl-64-full.tar.gz",
+                    "sha256": "c27d00dc83b5f443c35d785a0d22340f35c1cdb289c8cfe30417963475d4af29",
+                    "size": 207725021
+                },
                 {
                     "type": "file",
-                    "path": "placeholder.png"
+                    "path": "com.stencyl.Game-64.png"
                 },
                 {
                     "type": "file",
-                    "path": "com.stencyl.Game.desktop"
+                    "path": "com.stencyl.Game-128.png"
                 },
                 {
                     "type": "file",
-                    "path": "com.stencyl.Game.appdata.xml"
+                    "path": "com.stencyl.Game-512.png"
                 },
                 {
                     "type": "file",
-                    "path": "stencyl.sh"
+                    "path": "com.stencyl.Game.desktop"
                 },
                 {
                     "type": "file",
-                    "path": "firefox"
+                    "path": "com.stencyl.Game.appdata.xml"
```

Comment:
Not new in this patch but the appdata (presumably) refers to a "screenshot" which has embedded text and an OS X window border. I guess this is just inherited from upstream.

Reply:
Correct. I didn't see the need to address that in this PR (not least because it's a bit tricky for me as the app isn't 100% friendly to HiDPI)